filters/apimonitoring/filter.go

```diff
@@ -0,0 +1,158 @@
+package apimonitoring
+
+import (
+	"github.com/sirupsen/logrus"
+	"github.com/zalando/skipper/filters"
+	"net/http"
+	"regexp"
+	"strings"
+	"time"
+)
+
+const (
+	metricsSeparator = "."
+	metricCountAll   = "http_count"
+	metricCount500s  = "http5xx_count"
+	metricCount400s  = "http4xx_count"
+	metricCount300s  = "http3xx_count"
+	metricCount200s  = "http2xx_count"
+	metricLatency    = "latency"
+)
+
+const (
+	stateBagKeyPrefix = "filter.apimonitoring."
+	stateBagKeyState  = stateBagKeyPrefix + "state"
+)
+
+var (
+	unknownPath = &pathInfo{
+		ApplicationId: unknownElementPlaceholder,
+		ApiId:         unknownElementPlaceholder,
+		PathTemplate:  unknownElementPlaceholder,
+	}
+)
+
+type apiMonitoringFilter struct {
+	paths []*pathInfo
+}
+
+type pathInfo struct {
+	ApplicationId string
+	ApiId         string
+	PathTemplate  string
+	Matcher       *regexp.Regexp
+}
+
+// apiMonitoringFilterContext contains information about the metrics tracking
+// for one HTTP exchange (one routing). It serves to pass information from
+// the `Request` call to the `Response` call (stored in the context's `StateBag`).
+type apiMonitoringFilterContext struct {
+	// DimensionPrefix is the prefix to all metrics tracked during this exchange (generated only once)
+	DimensionsPrefix string
+	// Begin is the earliest point in time where the request is observed
+	Begin time.Time
+	// OriginalRequestSize is the initial requests' size, before it is modified by other filters.
+	OriginalRequestSize int64
+}
+
+func (f *apiMonitoringFilter) String() string {
+	return toJsonStringOrError(mapApiMonitoringFilter(f))
+}
+
+func (f *apiMonitoringFilter) Request(c filters.FilterContext) {
+	log := log.WithField("op", "request")
+
+	// Identify the dimensions "prefix" of the metrics.
+	dimensionsPrefix := f.getDimensionPrefix(c, log)
+
+	// Gathering information from the initial request for further metrics calculation
+	begin := time.Now()
+	originalRequestSize := c.Request().ContentLength
```

Comment:
Not needed.

Reply:
Fixed in dd090d4e62bfa893aa2daa69e60a35c635c11ce5